x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -135,6 +141,70 @@ export class OsqueryPlugin implements Plugin<OsqueryPluginSetup, OsqueryPluginSt
         await this.initialize(core, dataViewsService);
       }
 
+      let updatedPackageResult;
+      if (packageInfo && satisfies(packageInfo?.version ?? '', '<1.6.0')) {
```

Comment:
And also - as you mentioned doing this in here is probably not the best idea. I'll try set up a Task to handle it. Because now, it really is called only (and also every time) when kibana starts which makes it not applied when actually adding/upgrading the integration.

Reply:
Anyway, thanks for doing this, we definitely need this fix :)

Reply:
@tomsonpl

> is the trick to automatically update user's osquery integration ?

Yes, this is the idea.  We need to update the ES components that are installed by the osquery package.  That change went out here: https://github.com/elastic/integrations/pull/4990

We do NOT need to update individual integration policies.  All we need to do is upgrade the version of the osquery package that is installed.  This can be done with the install service as shown in the code above.  I tested this locally with above and can confirm that updating the integration policy is all that is needed to fix the issue.

> However, if we have integration installed on some agent policies, eg. we have policies with 1.5.1, and then use the script to upgrade the global integration package, the policies still have 1.5.1 - should this still fix the issue?

Yes, it will still fix the issue.  By upgrading the osquery integration package, we will install the needed pipeline to correct the problem of documents having the wrong value.  As stated above, I tested and confirmed that this works.

> And also - as you mentioned doing this in here is probably not the best idea. I'll try set up a Task to handle it. Because now, it really is called only (and also every time) when kibana starts which makes it not applied when actually adding/upgrading the integration.

Yes, I'd like to see if we could take it out of plugin start since it's a critical path.  However, we do only need to call this once at around the start of the application.  It should be a one time thing that updates the package and rolls over the indices, if needed.  After that, the problem should be fixed.

With the above, my thinking is that we use a task that is triggered and runs only once after the OSQuery plugin starts.  This would get the checks out of plugin start into a task that Kibana isn't dependent on to get started.  We don't want to tie to a user manual task such as updating a package policy, we'd rather do the fixes seamlessly when the user upgrades.

cc @patrykkopycinski @aleksmaus